packages/account/src/Components/api-token/_tests_/api-token-delete-button.spec.tsx

```diff
@@ -0,0 +1,99 @@
+        toggleOverlay: jest.fn(),
+    };
+    const mock_token = {
+        token: {
+            display_name: 'Token 1',
+            last_used: '12/31/2022',
+            scopes: ['read', 'trade'],
+            token: '1234567',
+        },
+    };
+
+    const renderAPIDeleteButton = () => {
+        render(
+            <ApiTokenContext.Provider value={mock_props}>
+                <ApiTokenDeleteButton {...mock_token} />
+            </ApiTokenContext.Provider>
+        );
+    };
+
+    it('should render ApiTokenDeleteButton', () => {
+        renderAPIDeleteButton();
+        expect(screen.getByTestId('dt_token_delete_icon')).toBeInTheDocument();
+        expect(screen.queryByText('Delete this token')).not.toBeInTheDocument();
+    });
+
+    it('should display Delete this token when mouse enter', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.hover(delete_icon);
+        expect(screen.getByText('Delete this token')).toBeInTheDocument();
+    });
+
+    it('should not Delete this token when mouse leave', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.hover(delete_icon);
+        expect(screen.getByText('Delete this token')).toBeInTheDocument();
+        userEvent.unhover(delete_icon);
+        expect(screen.queryByText('Delete this token')).not.toBeInTheDocument();
+    });
+
+    it('should display Popup when delete icon is clicked', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.click(delete_icon);
+        expect(screen.getByText('Delete token')).toBeInTheDocument();
+        expect(screen.getByText('Are you sure you want to delete this token?')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Yes, delete' })).toBeInTheDocument();
+    });
+
+    it('should close the modal when clicked on Cancel', async () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.click(delete_icon);
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        const cancel_button = screen.getByRole('button', { name: 'Cancel' });
+        userEvent.click(cancel_button);
+        await waitFor(() => expect(screen.queryByText('Delete token')).not.toBeInTheDocument());
+        expect(screen.queryByText('Are you sure you want to delete this token?')).not.toBeInTheDocument();
+        expect(screen.queryByRole('button', { name: 'Cancel' })).not.toBeInTheDocument();
+        expect(screen.queryByRole('button', { name: 'Yes, delete' })).not.toBeInTheDocument();
+    });
+
+    it('should should trigger deleteToken when clicked on Yes, delete', async () => {
```

Comment:
we are not using await anywhere in this test case. Do we need to wrap it async then?